File: StreamProviderWS/StreamProviderWS/Models/WebSocket/RequestsMessages/Video/WebRtc/PeerMessage.cs
using StreamProviderWS.Models.Common;

namespace StreamProviderWS.Models.WebSocket.RequestsMessages
{
    public class PeerMessage
    {
        public string By { get; set; }
        public string To { get; set; }
        public string RoomId { get; set; }
        public string Type { get; set; }
        public User User { get; set; }
        public object Ice { get; set; }
        public object Sdp { get; set; }
    }
}

File: VideoStreamerUI/src/app/entities/video-chat/peerMessage.ts
import { SocialUser } from 'angularx-social-login';
import { VideoInfo } from '../videoInfo';

export class PeerMessage {
    By: string;
    To: string;
    RoomId: string;
    Type: string;
    Ice: RTCIceCandidate;
    Sdp: RTCSessionDescriptionInit;
    User: SocialUser;
    VideoInfo: VideoInfo;

    constructor(by: string, to: string, roomId: string, type: string, user: SocialUser, videoInfo: VideoInfo, ice: RTCIceCandidate, sdp: RTCSessionDescriptionInit) {
        this.By = by;
        this.To = to;
        this.RoomId = roomId;
        this.Type = type;
        this.User = user;
        this.VideoInfo = videoInfo;
        this.Ice = ice;
        this.Sdp = sdp;
    }
}